### PR TITLE
Do not allow to run program as root

### PR DESCRIPTION
**Pull request by Amanse**
Simply check uid and euid and exit the process if it is found to be of root's.
One issue is checking these will cause build error for windows since crates do not implement getting uid and euid for windows target

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -19,6 +19,10 @@ use tracing_subscriber::{EnvFilter, Registry};
 
 #[tokio::main]
 async fn main() {
+    if sudo::check() == sudo::RunningAs::Root {
+        println!("Dura cannot be used as root");
```
Comment (tkellogg) [inline src/main.rs:23]:
```suggestion
        eprintln!("Dura cannot be used as root, to avoid data corruption");
```

**Review by tkellogg — APPROVED** [review]
Probably fine, but we may want to add a CLI option to allow null. Let's keep it this way for now though.

**Comment by tkellogg** [comment]
Doesn't look like `sudo` works on Windows. This might not be a problem on Windows. Maybe just make a pair of functions, one implemented for Windows (always return true) and one for all other platforms (use `sudo::check()`).

**Comment by Amanse** [comment]
that's exactly what i had thought but i slept😅 I'll push in a few hours

**Pull request closed without merging**